Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_flash_ex.c

```diff
@@ -719,84 +719,11 @@ uint32_t HAL_FLASHEx_GetSecInversion(void)
   *            @arg FLASH_BANK_2: Flash Bank 2
   *            @arg FLASH_BANK_BOTH: Flash Bank 1 and Bank 2
   * @retval HAL Status
+  * @warning Not supported on Microvisor.  Returns HAL_ERROR
   */
 HAL_StatusTypeDef HAL_FLASHEx_EnablePowerDown(uint32_t Banks)
 {
-  HAL_StatusTypeDef status = HAL_OK;
-  uint32_t tickstart;
-
-  /* Check the parameters */
-  assert_param(IS_FLASH_BANK(Banks));
-
-  /* Request power-down mode for Bank 1 */
-  if ((Banks & FLASH_BANK_1) != 0U)
-  {
-    /* Check PD1 and PDREQ1 bits (Bank 1 is not in power-down mode and not being
-       already under power-down request) */
-    if ((FLASH->NSSR & FLASH_NSSR_PD1) != 0U)
-    {
-      status = HAL_ERROR;
-    }
-    else if ((FLASH->ACR & FLASH_ACR_PDREQ1) != 0U)
-    {
-      status = HAL_ERROR;
-    }
-    else
-    {
-      /* Unlock PDREQ1 bit */
-      WRITE_REG(FLASH->PDKEY1R, FLASH_PDKEY1_1);
-      WRITE_REG(FLASH->PDKEY1R, FLASH_PDKEY1_2);
-
-      /* Set PDREQ1 in FLASH_ACR register */
-      SET_BIT(FLASH->ACR, FLASH_ACR_PDREQ1);
-
-      /* Check PD1 bit */
-      tickstart = HAL_GetTick();
-      while (((FLASH->NSSR & FLASH_NSSR_PD1) != FLASH_NSSR_PD1))
-      {
-        if ((HAL_GetTick() - tickstart) > FLASH_TIMEOUT_VALUE)
-        {
-          return HAL_TIMEOUT;
-        }
-      }
-    }
-  }
-
-  /* Request power-down mode for Bank 2 */
-  if ((Banks & FLASH_BANK_2) != 0U)
-  {
-    /* Check PD2 and PDREQ2 bits (Bank 2 is not in power-down mode and not being
-       already under power-down request) */
-    if ((FLASH->NSSR & FLASH_NSSR_PD2) != 0U)
-    {
-      status = HAL_ERROR;
-    }
-    else if ((FLASH->ACR & FLASH_ACR_PDREQ2) != 0U)
-    {
-      status = HAL_ERROR;
-    }
-    else
-    {
-      /* Unlock PDREQ2 bit */
-      WRITE_REG(FLASH->PDKEY2R, FLASH_PDKEY2_1);
-      WRITE_REG(FLASH->PDKEY2R, FLASH_PDKEY2_2);
-
-      /* Set PDREQ2 in FLASH_ACR register */
-      SET_BIT(FLASH->ACR, FLASH_ACR_PDREQ2);
-
-      /* Check PD2 bit */
-      tickstart = HAL_GetTick();
-      while (((FLASH->NSSR & FLASH_NSSR_PD2) != FLASH_NSSR_PD2))
-      {
-        if ((HAL_GetTick() - tickstart) > FLASH_TIMEOUT_VALUE)
-        {
-          return HAL_TIMEOUT;
-        }
-      }
-    }
-  }
-
-  return status;
+  return HAL_ERROR;
```

Comment:
Do you think HAL_ERROR is a better option to return here than HAL_OK?

Returning HAL_OK would make any porting exercise easier, whereas returning HAL_ERROR is likely to trigger an assertion or similar in the calling code, which in turn might be a good thing as then the programmer can remove the function call completely....

In all honesty, I'm undecided which is the best route to take, but I do think we should be consistent, and I think we should raise a ticket to ensure that all now benign functions do the same thing (like perhaps raise a linker warning)

Reply:
Yeah there's some inconsistency here. I decided to make `RCC_SetFlashLatencyFromMSIRange` return `HAL_OK` because I think the goal of the caller of that function is to have the flash latency set correctly, and we (in the kernel) are doing that. Whereas a caller of say `HAL_FLASHEx_EnablePowerDown ` has a different goal, and one that we can't really say has been achieved by doing nothing, so that's why that (and others) returns `HAL_ERROR`. Can definitely see the arguments both ways though.